### PR TITLE
Quote exclusion regexes properly (#113)

### LINKED ISSUE #113: Ignore entries are regexes, not globs
In the [example config](https://github.com/rackerlabs/lambda-uploader/blob/master/example/lambda.json) and elsewhere the `ignore` section is treated like a list of filesystem globs (e.g. `*.txt`), whereas [the code](https://github.com/rackerlabs/lambda-uploader/blob/0f6b957ee19d790283dc7f150f9a9ef0c49422d0/lambda_uploader/utils.py#L62) suggests these are in fact regular expressions matching on the path.

The examples all work as intended (largely because `r'.'` matches the character `.`) but would also match unwanted (albeit strange) files, e.g.  `Xgit` or `/Xpyc`.

Elsewhere in the code, arbitrary filenames should probably be escaped when added to this list, in case they contain special characters (e.g. `.` but also `[]()-*?` etc) and start matching incorrectly.

I've got (most of) a patch for this, if that helps.

### PR DESCRIPTION
**Pull request by declension**
* Regex-quote arbitrary files added to exclusion list
 * Add tests around this behaviour demonstrating regex nature
 * Update examples and docs to demonstrate this
 * Use immutable keyword parameters defaults too

Fixes #113

### CODE REVIEW

File: lambda_uploader/package.py [hunk 1]
```diff
@@ -223,6 +224,7 @@ def package(self, ignore=[]):
             those files when creating the zip file. The paths to be matched are
             local to the source root.
         """
+        ignore = ignore or []
```
Comment (jarosser06) [inline lambda_uploader/package.py]: I don't understand why this was added in favor over setting the default for the ignore arg to [].

File: lambda_uploader/utils.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 LOG = logging.getLogger(__name__)
 
 
-def copy_tree(src, dest, ignore=[], include_parent=False):
+def copy_tree(src, dest, ignore=None, include_parent=False):
```
Comment (jarosser06) [inline lambda_uploader/utils.py]: Again, this seems like an extra step that the default arg already covers.

**Comment by declension** [comment]
Sure: mutable default arguments can often be a source bugs in Python, and some IDEs warn about their usage. [This book / guide](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments) has the best explanation / example I've seen.

**Comment by jarosser06** [comment]
I've never run into that. Thanks!

**Comment by declension** [comment]
Thank you

**Pull request merged**